### PR TITLE
Transfer subscriptions trial

### PR DESCRIPTION
**Pull request by tspenov**
## Changes

1. Refactoring subscriptions API.
2. Transfer trial

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have tried to find clearer solution before commenting hard-to-understand parts of code
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: lib/sanbase/billing/billing.ex [hunk 1]
```diff
@@ -70,4 +72,27 @@ defmodule Sanbase.Billing do
       {:error, _} -> true
     end)
   end
+
+  @spec create_or_update_stripe_customer(%User{}, String.t() | nil) ::
+          {:ok, %User{}} | {:error, %Stripe.Error{}}
+  def create_or_update_stripe_customer(_, _card_token \\ nil)
```
Comment (IvanIvanoff) [inline lib/sanbase/billing/billing.ex:78]: Here you do not need to underscore vars. Let's keep the proper names for better documentation:
```suggestion
  def create_or_update_stripe_customer(user, card_token \\ nil)
```

File: lib/sanbase/billing/billing.ex [hunk 2]
```diff
@@ -70,4 +72,27 @@ defmodule Sanbase.Billing do
...
+
+  def create_or_update_stripe_customer(%User{stripe_customer_id: stripe_id} = user, card_token)
+      when is_nil(stripe_id) do
```
Comment (IvanIvanoff) [inline lib/sanbase/billing/billing.ex:81]:
```suggestion
  def create_or_update_stripe_customer(%User{stripe_customer_id: nil} = user, card_token) do
```

File: lib/sanbase/billing/billing.ex [hunk 3]
```diff
@@ -70,4 +72,27 @@ defmodule Sanbase.Billing do
...
+
+  def create_or_update_stripe_customer(%User{stripe_customer_id: stripe_id} = user, nil)
+      when is_binary(stripe_id) do
+    {:ok, user}
+  end
+
+  def create_or_update_stripe_customer(%User{stripe_customer_id: stripe_id} = user, card_token)
+      when is_binary(stripe_id) do
+    with {:ok, _} <- StripeApi.update_customer(user, card_token) do
+      {:ok, user}
```
Comment (IvanIvanoff) [inline lib/sanbase/billing/billing.ex:95]: Should we force preload something here?

Comment (tspenov) [reply lib/sanbase/billing/billing.ex:108]: No, here we update customer in Stripe if we have new card but the database user stays the same and we return it.

**Pull request merged**